### PR TITLE
fix(openlr): Fixes km to meter conversion in OpenLr serializer

### PR DESCRIPTION
**Pull request by PureW**
There was a bug in the OpenLr serialization leading to the `dnp` value being wrong. A value in kilometer was treated as meter.

The change also renames TripLeg.Edge.length to TripLeg.Edge.length_km to avoid this issue in the future.

### CODE REVIEW

File: test/mapmatch.cc [hunk 1]
```diff
@@ -1428,7 +1428,7 @@ TEST(Mapmatch, openlr_parameter_true_osrm_api) {
       "CwOduyULYiKJAAAV//0iGw==",
       "CwOdxCULYCKJAAAN//4iGw==",
       "CwOdySULXyKJAAAf//EiGw==",
-      "CwOd1yULVyKLAABV/9AiGw==",
+      "CwOd1yULVyKLAQBV/9AiGw==",
```
Comment (danpaz) [inline test/mapmatch.cc:1431]: Was this change intended for this PR?

Comment (PureW) [reply test/mapmatch.cc:1431]: I just updated the description/name of the PR. It actually fixes a bug in the openlr serialization at https://github.com/valhalla/valhalla/pull/2688/files#diff-dd901075ccbb89ba7900fb2348eac4f9da77488c897257ba9f657a10603531a8R80

The rename (most of the diff) is only made to avoid a similar issue in future.

Comment (PureW) [reply test/mapmatch.cc:1431]: So yes, this change in test-value is expected.

Comment (mandeepsandhu) [reply test/mapmatch.cc:1431]: Yeah I had the same confusion. I missed the line you pointed above, thinking the diff was simply a `s/length/length_km` 🙂

**Review by mandeepsandhu — DISMISSED** [review]

File: test/mapmatch.cc (hunk 1 above)
Comment (danpaz) [reply test/mapmatch.cc:1431]: 👍

**Review by mandeepsandhu — APPROVED** [review]
